riptide-stream/src/main/java/org/zalando/riptide/stream/Streams.java

```diff
@@ -0,0 +1,114 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import org.springframework.http.converter.HttpMessageConverter;
+import org.zalando.riptide.ThrowingConsumer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.reflect.TypeParameter;
+import com.google.common.reflect.TypeToken;
+
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * Main entry point for <b>Riptide Streams</b> extension to capture arbitrary infinite object streams. It allows to
+ * receive infinite streams using application/x-json-stream and application/json-seq format, as well as simple finite
+ * streams from lists and arrays. Must be enable by registering the {@link StreamConverter} with Riptide (using
+ * {@link Streams#streamConverter()}) and declare a route for your stream that is calling a the stream consumer as
+ * follows:
+ * 
+ * <pre>
+ * try (Rest rest = Rest.builder().baseUrl("https://api.github.com").converter(streamConverter()).build()) {
+ *     rest.get("/repos/{org}/{repo}/contributors", "zalando", "riptide")
+ *             .accept(MediaType.APPLICATION_JSON)
+ *             .dispatch(series(),
+ *                     on(SUCCESSFUL).call(streamOf(User.class),
+ *                             forEach(user -> println(user.login + " (" + user.contributions + ")"))))
+ *             .get();
+ * }
+ * </pre>
+ * 
+ * <b>Note:</b> The stream converter is an replacement to the default spring JSON converter that does not support
+ * streaming, and thus should be not registered together with it.
+ */
+public final class Streams {
+
+    public static <T> TypeToken<Stream<T>> streamOf(final Class<T> type) {
+        return streamOf(TypeToken.of(type));
+    }
+
+    @SuppressWarnings("serial")
+    public static <T> TypeToken<Stream<T>> streamOf(final TypeToken<T> type) {
+        final TypeToken<Stream<T>> listType = new TypeToken<Stream<T>>() {
+        };
+
+        final TypeParameter<T> elementType = new TypeParameter<T>() {
+        };
+
+        return listType.where(elementType, type);
+    }
+
+    @SuppressWarnings("serial")
+    private static class UncheckedConsumerException extends RuntimeException {
+
+        public UncheckedConsumerException(Exception cause) {
+            super(cause);
+        }
+
+        @Override
+        public Exception getCause() {
+            return (Exception) super.getCause();
+        }
+    }
+
+    public static <I> ThrowingConsumer<Stream<I>> forEach(final ThrowingConsumer<I> consumer) {
+        return (input) -> {
+            try {
+                if (input == null) {
+                    return;
+                }
+                input.forEach(wrap(consumer));
+            } catch (UncheckedConsumerException ex) {
+                throw ex.getCause();
+            }
+        };
+    }
+
+    private static <I> Consumer<? super I> wrap(final ThrowingConsumer<I> consumer) throws Exception {
```

Comment:
But it's easy to see where the UncheckedConsumerException is thrown/catched and why.